src/environment/shell.go

```diff
@@ -319,6 +320,25 @@ func (env *ShellEnvironment) HasFilesInDir(dir, pattern string) bool {
 	return len(matches) > 0
 }
 
+func (env *ShellEnvironment) HasFileInParentDirs(pattern string, depth uint) bool {
+	defer env.trace(time.Now(), "HasFileInParent", pattern, fmt.Sprint(depth))
+	currentFolder := env.Pwd()
+
+	for c := 0; c < int(depth); c++ {
```

Comment:
why uint and then convert to int here?

Reply:
When used without type conversion there is an error:
_cannot compare c < depth (mismatched types int and uint)compiler[MismatchedTypes]_
uint in the parameter seemed to me more accurate, carrying an intention of passing only a non-negative number. 
But I'm a fresher in Go world, with habits from my long-used languages, please advice what is the best or correct from your point of view.

Reply:
Should it be rather
`for c := uint(0); c < depth; c++ {`
?